### PR TITLE
Strip obsolete constraints

### PR DESCRIPTION
**Pull request by dabrahams**

### CODE REVIEW

**Comment by dabrahams** [comment]
@swift-ci Please test and merge

**Comment by dabrahams** [comment]
@swift-ci Please test and merge

**Comment by dabrahams** [comment]
@swift-ci Please test and merge

File: test/Prototypes/PatternMatching.swift [hunk 1]
```diff
@@ -35,18 +35,16 @@ protocol Pattern {
   associatedtype MatchData = ()
   
   func matched<C: Collection>(atStartOf c: C) -> MatchResult<Index, MatchData>
-  where C.Index == Index, Element_<C> == Element
+  where C.Index == Index, C.Element == Element
```
Comment (slavapestov) [inline test/Prototypes/PatternMatching.swift]: There are other usages of `Element_` remaining, are they still needed?

**Review by milseman — COMMENTED** [review]
As Slava mentions, is it possible to remove Element_?

File: test/Prototypes/PatternMatching.swift (hunk 1 above)
Comment (dabrahams) [reply test/Prototypes/PatternMatching.swift]: Good point, @slavapestov!  I'll address those too.

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

File: test/Prototypes/PatternMatching.swift (hunk 1 above)
Comment (dabrahams) [reply test/Prototypes/PatternMatching.swift]: Done.

**Comment by dabrahams** [comment]
@swift-ci Please smoke test and merge

**Comment by dabrahams** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**